File: C#-RelojChecador/ProyectoRelojChecador/ChecadaQuery.cs
using System;
using System.Collections.Generic;
using System.Data.SqlClient;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace ProyectoRelojChecador
{
    public class ChecadaQuery
    {
        //REGISTRAR CHECADA 
        public static int AgregarChecada(Checada checada) 
        {
            int retorno = 0;



            using (SqlConnection conexion = BDPrincipal.obtenerConexion()) 
            {

                string query = "INSERT INTO CHECADA (ID_EMPLEADO_TC,HORA,FECHA) VALUES("
                    + checada.id + ",'" + checada.hora + "','" + checada.fecha + "');";

                SqlCommand comando = new SqlCommand(query, conexion);

                retorno = comando.ExecuteNonQuery();

            }

            return retorno;
        
        }//FIN DE AGREGAR CHACADAS


        public static int AgregarChecada2(Checada checada)
        {
            int retorno = 0;


                using (SqlConnection conexion = BDPrincipal.obtenerConexion())
            {

                string query = "INSERT INTO CHECADA (ID_EMPLEADO_TC,HORA,FECHA) VALUES("
                    + checada.id + ",'" + checada.hora + "','" + checada.fecha + "');";

                SqlCommand comando = new SqlCommand(query, conexion);

                retorno = comando.ExecuteNonQuery();

            }

            return retorno;

        }//FIN DE AGREGAR CHACADAS


        //ELIMINAR EMPLEADO 
        public static int eliminarChecadaEmpleado(int parametroid) 
        {
            int retorno = 0;

            using (SqlConnection conexion = BDPrincipal.obtenerConexion()) 
            {
                string query = "DELETE FROM CHECADA WHERE ID_EMPLEADO_TC = "+parametroid+"";

                SqlCommand comando = new SqlCommand(query, conexion);

                retorno = comando.ExecuteNonQuery();

            }

            return retorno;

        }// END ELIMANR EMPLADO

        //PRUEBAS /////////////////////////////////////////////////////////////////////////////////

        public static int eliminarChecadaBonoEmpleado(int parametroid)
        {
            int retorno = 0;

            using (SqlConnection conexion = BDPrincipal.obtenerConexion())
            {
                string query = "DELETE FROM CHECADABONO WHERE ID_EMPLEADO_TB = " + parametroid + "";

                SqlCommand comando = new SqlCommand(query, conexion);

                retorno = comando.ExecuteNonQuery();

            }

            return retorno;

        }// END ELIMANR EMPLADO



        public static int ContarChecadas(int id)
        {
            int retorno = 0;

            using (SqlConnection conexion = BDPrincipal.obtenerConexion())
            {
                string fechaActual = DateTime.Now.ToString("yyyy-MM-dd");

                string query = @"SELECT COUNT(*) 
                         FROM CHECADA 
                         WHERE ID_EMPLEADO_TC = @id
                         AND FECHA = @fecha";

                SqlCommand comando = new SqlCommand(query, conexion);

                comando.Parameters.AddWithValue("@id", id);
                comando.Parameters.AddWithValue("@fecha", fechaActual);

                retorno = Convert.ToInt32(comando.ExecuteScalar());
            }

            return retorno;
        }//fin

        public static int AgregarChecadaBono(int id,String fecha,int indicador)
        {
            int retorno = 0;

            using (SqlConnection conexion = BDPrincipal.obtenerConexion())
            {
                string query = "INSERT INTO CHECADABONO (ID_EMPLEADO_TB,FECHA,BONO) VALUES(" + id + ",'" + fecha + "'," + indicador + ");";

                SqlCommand comando = new SqlCommand(query, conexion);

                retorno = comando.ExecuteNonQuery();

            }

            return retorno;

        }//FIN DE AGREGAR CHACADAS






    }//FIN DE MI CLASE
}

File: Java-RolojChecador/src/app/ChecadaQuery.java

package app;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import javax.swing.JOptionPane;

public class ChecadaQuery 
{
    public void AgregarChecada(Checada checada)
    {
            
        try
        {
            Connection con = BDPrincipal.obtenerConexion();
            PreparedStatement pa = con.prepareStatement(
            "INSERT INTO CHECADA (ID_EMPLEADO_TC, HORA, FECHA) VALUES (?, ?, ?)"
            );
            
            pa.setInt(1, checada.getId());        
            pa.setString(2, checada.getHora());      
            pa.setString(3, checada.getFecha());        
            pa.executeUpdate();
            JOptionPane.showMessageDialog(null, "Haz Checado");
        
        }catch(SQLException e)
              {
                  JOptionPane.showMessageDialog(null, e.toString());
              }
        
    }//FIN DE MI AGREGAR CHECADA
    
    //ELIMINAR CHECADA 
    
    public void EliminarChecada(int parametroid)
     {
         ResultSet resultado = null;
        

    try {
        // Preparar conexión y consulta
        Connection conexion = BDPrincipal.obtenerConexion();
        String consulta = "DELETE FROM CHECADA WHERE ID_EMPLEADO_TC = ?";
        PreparedStatement sql = conexion.prepareStatement(consulta);

        // Asignar valor al parámetro
        sql.setInt(1, parametroid);

        // Ejecutar la consulta
        resultado = sql.executeQuery();

    } catch (SQLException ex) {
       // JOptionPane.showMessageDialog(null, ex.toString());
    }
     
     }//FIN DE MI FUNCION 
    
    /////PRUEBA
    
    public void EliminarChecadaBono(int parametroid)
     {
         ResultSet resultado = null;
        

    try {
        // Preparar conexión y consulta
        Connection conexion = BDPrincipal.obtenerConexion();
        String consulta = "DELETE FROM CHECADABONO WHERE ID_EMPLEADO_TB = ?";
        PreparedStatement sql = conexion.prepareStatement(consulta);

        // Asignar valor al parámetro
        sql.setInt(1, parametroid);

        // Ejecutar la consulta
        resultado = sql.executeQuery();

    } catch (SQLException ex) {
       // JOptionPane.showMessageDialog(null, ex.toString());
    }
     
     }//FIN DE MI FUNCION 
    
    public void AgregarChecadaBono(int id,String fecha,int bono)
    {
            
        try
        {
            Connection con = BDPrincipal.obtenerConexion();
            PreparedStatement pa = con.prepareStatement(
            "INSERT INTO CHECADABONO (ID_EMPLEADO_TB,FECHA,BONO) VALUES (?, ?, ?)"
            );
            
            pa.setInt(1, id);        
            pa.setString(2, fecha);      
            pa.setInt(3, bono);        
            pa.executeUpdate();
            //JOptionPane.showMessageDialog(null, "Haz Checado");
        
        }catch(SQLException e)
              {
                  JOptionPane.showMessageDialog(null, e.toString());
              }
        
    }//FIN DE MI AGREGAR CHECADA BONO 
    
    /////////////////////
    
     public int ContarChecadas(int id) {
        int retorno = 0;
        Connection conexion = null;
        PreparedStatement comando = null;
        ResultSet resultado = null;

        try {
            // Obtener conexión
            conexion = BDPrincipal.obtenerConexion(); // Asegúrate de implementar este método
            String query = "SELECT COUNT(*) FROM CHECADA WHERE ID_EMPLEADO_TC = ?";
            comando = conexion.prepareStatement(query);
            comando.setInt(1, id); // Configurar el parámetro

            // Ejecutar la consulta
            resultado = comando.executeQuery();
            if (resultado.next()) {
                retorno = resultado.getInt(1); // Obtener el primer valor de la primera fila
            }
        } catch (SQLException e) {
            e.printStackTrace(); // Manejo de excepciones
        } finally {
            // Cerrar recursos
            try {
                if (resultado != null) resultado.close();
                if (comando != null) comando.close();
                if (conexion != null) conexion.close();
            } catch (SQLException e) {
                e.printStackTrace();
            }
        }

        return retorno;
    }
    
    
    
    
    
}//FIN DE MI CLASE 


